tests/model/test_core.py

```diff
@@ -527,6 +527,33 @@ def test_model_var_maps():
     assert model.rvs_to_transforms[x] is None
 
 
+class TestTransformArgs:
+    def test_transform_warning(self):
+        with pm.Model():
+            with pytest.warns(
+                UserWarning,
+                match="To disable default transform,"
+                " please use default_transform=None"
+                " instead of transform=None. Setting transform to"
+                " None will not have any effect in future.",
+            ):
+                a = pm.Normal("a", transform=None)
+
+    def test_transform_order(self):
+        with pm.Model() as model:
+            x = pm.Normal("x", transform=Interval(0, 1), default_transform=log)
+        assert isinstance(model.rvs_to_transforms[x].transform_list[0], LogTransform)
+        assert isinstance(model.rvs_to_transforms[x].transform_list[1], Interval)
+
+    def test_default_transform_is_applied(self):
+        with pm.Model() as model1:
+            x1 = pm.LogNormal("x1", 0, 1, transform=Interval(-2, 2), default_transform=None)
+        with pm.Model() as model3:
+            x2 = pm.LogNormal("x2", 0, 1, transform=Interval(-2, 2), default_transform=log)
```

Comment:
This is fine but is not very realistic. What about ordered transform + log which is the example that motivated this PR?

```suggestion
            x2 = pm.LogNormal("x2", 0, 1, transform=Interval(-2, 2), default_transform=log)
```

Reply:
The rest of the logic of the test is spot-on!